common/src/main/java/com/wynntils/features/user/ChatMentionFeature.java

```diff
@@ -0,0 +1,129 @@
+
+    @Override
+    protected void onConfigUpdate(ConfigHolder configHolder) {
+        pattern = Pattern.compile(
+                "(?<!\\[)\\b(" + McUtils.mc().getUser().getName()
+                        + (aliases.length() > 0 ? "|" + aliases.replace(",", "|") : "") + ")\\b(?!:|])",
+                Pattern.CASE_INSENSITIVE);
+    }
+
+    @SubscribeEvent()
+    public void onChat(ChatMessageReceivedEvent e) {
+        Component message = e.getMessage();
+
+        Matcher looseMatcher = pattern.matcher(ComponentUtils.getUnformatted(message));
+
+        if (looseMatcher.find()) {
+            if (markMention) {
+                e.setMessage(rewriteComponentWithHighlight(message));
+            }
+            if (dingMention) {
+                McUtils.playSound(SoundEvents.NOTE_BLOCK_PLING.value());
+            }
+        }
+    }
+
+    private Component rewriteComponentWithHighlight(Component comp) {
+        MutableComponent curr = MutableComponent.create(comp.getContents()).withStyle(comp.getStyle());
+        // .getString() is used here as it gives formattingchars when those exist. It is needed for guild messages
+        // because wynn still uses legacy coloring for it.
+        String text = curr.getString();
+
+        if (text == "") {
+            // case: component has no text -> pass the component thru and process its siblings
+            List<Component> sib = comp.getSiblings();
+            if (sib.size() == 0) {
+                // no siblings -> this is an end component return self
+                return comp;
+            } else {
+                // some siblings -> check them
+                for (Component c : sib) {
+                    curr.append(rewriteComponentWithHighlight(c));
+                }
+            }
+        } else {
+            // component has text -> check if it has the mention
+            Matcher match = pattern.matcher(text);
+
+            int nextStart = 0;
+
+            List<MutableComponent> comps = new ArrayList();
+
+            String lastcol = "";
+
+            // NOTE: Message is modified here!
+            while (match.find()) {
+                // Mention found -> split component and mark the
+                String before = text.substring(nextStart, match.start());
+                String name = text.substring(match.start(), match.end());
+
+                comps.add(Component.literal(before).withStyle(comp.getStyle()));
+                // styling is not used as it breaks guild chat because wynn in their infinite wisdom decided to make
+                // guild chat not use the propper styling
+                comps.add(Component.literal("§e" + name));
+
+                nextStart = match.end();
+            }
+
+            MutableComponent afterComp =
+                    Component.literal(text.substring(nextStart)).withStyle(comp.getStyle());
+
+            // process any siblings
+            List<Component> sib = comp.getSiblings();
+            if (sib.size() != 0) {
```

Comment:
the if check is unecessary, the for loop will not do anything if there are no siblings.